components/styled/table/src/molecules/questionnaire-table.tsx

```diff
@@ -23,24 +25,76 @@ export const mapQuestionnaireObjectsToVerticalTableData = (
       accessor: record?.id ?? '',
     })),
   ],
-  rows: definitionItems.map((def) => ({
-    id: def?.linkId ?? '',
-    cells: [
-      {
-        key: 'property',
-        value: def?.text ?? '',
-      },
-      ...records.map((record) => ({
-        key: record.id,
-        value: EnsureMaybe<string>(
-          record.item?.find((item) => item?.linkId === def?.linkId)?.answer?.find((answer) => !!answer)?.valueString,
-          ''
-        ),
-      })),
-    ],
-  })),
+  rows: buildVerticalCellRowsRecursive(definitionItems, records),
 })
 
+const buildVerticalCellRowsRecursive = (
+  definitionItems: QuestionnaireItem[],
+  records: QuestionnaireResponse[]
+): CellRow[] =>
+  definitionItems.map((def) => {
+    const containsSubRows: boolean = (def?.item && def.item.length > 0) ?? false
+    const subRows = containsSubRows
+      ? buildVerticalCellRowsRecursive(EnsureMaybe(def?.item?.map((x) => EnsureMaybe(x))), records)
+      : []
+
+    return {
+      id: def?.linkId ?? '',
+      cells: [
+        {
+          key: 'property',
+          value: def?.text ?? '',
+        },
+        ...records.map((record) => ({
+          key: record.id,
+          value: containsSubRows ? (
+            <Checkbox
+              size="small"
+              color="primary"
+              checked={subRows.some((x) => x.cells.find((y) => y.key === record.id)?.value !== '')}
+              style={{ padding: 0 }}
+            />
+          ) : (
+            findQuestionnaireResponseAnswerValue(EnsureMaybe(def?.linkId), record?.item ?? [])
+          ),
+        })),
+      ],
+      subRows,
+    }
+  }) ?? []
+
+const findQuestionnaireResponseAnswerValue = (id: string, items: Maybe<Maybe<QuestionnaireResponseItem>>[]): string => {
+  const answerItem = findAnswerByLinkIdRecursive(id, items)
+  return answerItem ? EnsureMaybe<string>(answerItem?.answer?.find((x) => !!x)?.valueString, '') : ''
+}
+
+const findAnswerByLinkIdRecursive = (
```

Comment:
You can shorten this method quite a lot by replacing the old form `for` loop with a `forEach` then swapping each `break` statement for a return. You also don't need to declare the `itemFound` variable then either  :)

But I'd be wary of how you're using `EnsureMaybe`. Currently you're initialising `defaultAnswer` as the result of two `EnsureMaybe` calls and then checking its truthiness on the next line, but its truthiness is already guaranteed by the `EnsureMaybe` (unless you give it a default value)

Reply:
I have refactored code around the default answer. I'm still using the `for` loop as it allows iteration through all the sub lists with in the list even when no item found where as `forEach `will require return statement for all paths regardless of whether the item is to be found or not. `for` gives us the flexibility to recursively go through lists within lists and break out of loops when item is found at will and in case of not finding item it will go through every list with in the record set and return undefined finally.